src/inmanta/logging.py

```diff
@@ -178,14 +153,18 @@ def custom_log_record_factory(
         This log record factory makes sure that the name of the log record is updated
         in the following way while executing in the "compiler" or "exporter" logger mode:
 
-        * The name of the Inmanta module: When the log record was created by a source file in an Inmanta module.
+        * The name of the Inmanta module: When the log record was created by a source file in an Inmanta module
+                                          and the name of the logger was set to __name__.
         * compiler: When executing in compiler mode and the log record doesn't come from an Inmanta module.
         * exporter: When executing in exporter mode and the log record doesn't come from an Inmanta module.
         """
         new_logger_name: str
-        if self._logger_mode is LoggerMode.COMPILER or self._logger_mode is LoggerMode.EXPORTER:
-            inmanta_module_name: Optional[str] = self._get_module_name_for_source_file(pathname)
-            new_logger_name = inmanta_module_name if inmanta_module_name else self._logger_mode.value
+        if not self._keep_logger_names and self._logger_mode in [LoggerMode.COMPILER, LoggerMode.EXPORTER]:
+            match: Optional[re.Match[str]] = self._inmanta_plugin_pkg_regex.match(name)
+            if match:
+                new_logger_name = match.groupdict()["module_name"]
+            else:
+                new_logger_name = self._logger_mode.value
```

Comment:
There is still the possibility of this being a third party warning, in which case I think we want to keep the original. There is also the case of what is currently the `inmanta.pip` logger that would now become `compiler`, right?

Reply:
The question is what we want to achieve by changing the logger names of the log records. I understood the intention was to make it clear to the end user whether we are currently executing the compile or exporter stage. From that perspective it's desired behavior to log everything that comes from the `inmanta.pip` logger as `compiler`. But maybe I misunderstood the goal of this ticket. Do you have something else in mind?

Reply:
I'm not sure to be honest. But I seem to recall we introduced the artificial `inmanta.pip` logger explicitly to make it clear that this came from the pip subprocess. I find it strange to put it back under the `compiler` banner but admittedly I've never been very good at this user-oriented stuff so I also don't want to press my opinion too much. Perhaps @bartv has an opinion on this?

Reply:
I think `pip` would be a good idea for that output. Because it also is not compiler.